### PR TITLE
Make transit import command work for the transform backend

### PR DESCRIPTION
**Pull request by DeLuci**

### CODE REVIEW

File: command/transit_import_key.go [hunk 1]
```diff
@@ -177,6 +185,21 @@ func importKey(c *BaseCommand, operation string, flags *FlagSets, args []string)
 	}
 }
 
+func extractPathInfo(path string) (string, string, string, string, error) {
+	parts := transitPath.FindStringSubmatch(path)
+	if len(parts) == 3 {
+		return "transit", parts[1], parts[2], "", nil
+	}
+
+	parts = transformPath.FindStringSubmatch(path)
```
Comment (sgmiller) [inline command/transit_import_key.go:194]: So unfortunately secrets engines can be mounted at any path.  We can count on the paths after the mount path but we can't deduce which type of backend we're targetting.

Instead, we should create new transform versions of the cli commands that share almost all the same logic (reuse code) but don't have to guess whether it's transit or transform.

**Review by sgmiller — CHANGES_REQUESTED** [review]

File: command/transit_import_key.go (hunk 1 above)
Comment (stevendpclark) [reply command/transit_import_key.go:194]: So you are disagreeing on this PR to make it more explicit to the end-user they want to use transform from the cli point of view, as this is relying on paths after the mount path to identify what engine the end-user is targeting.

**Review by victorr — APPROVED** [review]
💯

**Review by sgmiller — APPROVED** [review]
Good work!

**Comment by victorr** [comment]
The vault-enterprise PR for this changes is: https://github.com/hashicorp/vault-enterprise/pull/4075

That PR adds unit tests for the new commands.

**Pull request merged**